proof/aggregate_in_g1.saw

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2020 Galois, Inc.
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+*/
+
+// NOTE: we only consider inputs that either have the infinity-bit set or compressed-bit set
+
+// aggregate_in_g1_error_precond describes the circumstances under which aggregate_in_g1 returns an error (assuming either the compressed bit is set of the infinity bit is set)
+let {{
+  aggregate_in_g1_error_precond : [48][8] -> Bool
+  aggregate_in_g1_error_precond zwire =
+      ~(zwire@0)@1 // not infinity
+    /\ (   uncompress_E1_x zwire >= `p
+        \/ ~ is_square_fp (uncompress_E1_y2 zwire) // not on curve
+        \/ uncompress_E1_x_fp zwire == Fp.field_zero // uncompresses to zero
+        \/ ~ is_in_g1_impl (uncompress_E1_OK zwire) ) // not in group g1
+}};
+
+// case in which the input is not null and we return BLST_SUCCESS:
+let blst_aggregate_in_g1_OK_nonnull_spec = do {
+  //
+  (out, out_ptr) <- ptr_to_fresh "blst_aggregate_in_g1_out" (llvm_struct "struct.POINTonE1");
```

Comment:
Does out_ptr really need to point to initialized memory here?  It does not appear to be read in the code, only written to, sometimes.